ams/ams-server/src/main/java/com/netease/arctic/ams/server/optimize/IcebergFullOptimizePlan.java

```diff
@@ -88,9 +88,17 @@ protected boolean partitionNeedPlan(String partitionToPath) {
         TableProperties.SELF_OPTIMIZING_MAJOR_TRIGGER_DUPLICATE_RATIO_DEFAULT);
     // delete files total size reach target_size * duplicate_ratio
     if (deleteFilesTotalSize > targetSize * duplicateRatio) {
-      LOG.debug("{} ==== need native Full optimize plan, partition is {}, " +
+      LOG.debug("{} ==== need native Full optimize plan by {} > {} * {} , partition is {}, " +
               "delete files totalSize is {}, target size is {}",
-          tableId(), partitionToPath, deleteFilesTotalSize, targetSize);
+          tableId(), deleteFilesTotalSize, targetSize, duplicateRatio,
+        partitionToPath, deleteFilesTotalSize, targetSize);
+      return true;
+    }
+
+    if (checkOptimizeInterval(partitionToPath)) {
```

Comment:
Thank you for your suggestion, but I would like to ask how I can obtain whether the current partition is healthy.

Reply:
For example, if a certain partition was last fully optimized and no data has been written since then, will it be fully optimized again at the next scheduled time?

Reply:
This sounds good. After the last full optimization on a partition, if there are no new data writes, it is important to avoid full optimization again.

And I would like to ask how it would be better to determine if there is any new data being written.I am relatively lacking in knowledge in this area. Thank you.

Reply:
You can determine whether there have been any data changes in a partition by checking whether it has any "delete" files or has two or more small files.

Reply:
Thx，Let me confirm.

Reply:
Although this may cause partitions without delete files to not trigger full optimizing, it is good enough to solve most of the problems in the scenario.